src/single-line-diagram-viewer.ts

```diff
@@ -0,0 +1,608 @@
+/**
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import { Point, SVG, Svg, ViewBoxLike } from '@svgdotjs/svg.js';
+import '@svgdotjs/svg.panzoom.js';
+
+type DIMENSIONS = { width: number; height: number; viewbox: VIEWBOX };
+type VIEWBOX = { x: number; y: number; width: number; height: number };
+
+const SVG_NS = 'http://www.w3.org/2000/svg';
+
+const SWITCH_COMPONENT_TYPES = new Set([
+    'BREAKER',
+    'DISCONNECTOR',
+    'LOAD_BREAK_SWITCH',
+]);
+
+const FEEDER_COMPONENT_TYPES = new Set([
+    'LINE',
+    'LOAD',
+    'BATTERY',
+    'DANGLING_LINE',
+    'GENERATOR',
+    'VSC_CONVERTER_STATION',
+    'LCC_CONVERTER_STATION',
+    'HVDC_LINE',
+    'CAPACITOR',
+    'INDUCTOR',
+    'STATIC_VAR_COMPENSATOR',
+    'TWO_WINDINGS_TRANSFORMER',
+    'TWO_WINDINGS_TRANSFORMER_LEG',
+    'THREE_WINDINGS_TRANSFORMER',
+    'THREE_WINDINGS_TRANSFORMER_LEG',
+    'PHASE_SHIFT_TRANSFORMER',
+]);
+
+const ARROW_SVG =
+    '<svg width="44" height="44" viewBox="0 0 44 44" fill="none" xmlns="http://www.w3.org/2000/svg"><path class="arrow" fill-rule="evenodd" clip-rule="evenodd" d="M16 24.0163L17.2358 25.3171L21.9837 20.5691L26.7317 25.3171L28 24.0163L21.9837 18L16 24.0163Z"/></svg>';
+const ARROW_HOVER_SVG =
+    '<svg width="44" height="44" viewBox="0 0 44 44" fill="none" xmlns="http://www.w3.org/2000/svg"><path class="arrow-hover" fill-rule="evenodd" clip-rule="evenodd" d="M22 35C29.1797 35 35 29.1797 35 22C35 14.8203 29.1797 9 22 9C14.8203 9 9 14.8203 9 22C9 29.1797 14.8203 35 22 35ZM17.2358 25.3171L16 24.0163L21.9837 18L28 24.0163L26.7317 25.3171L21.9837 20.5691L17.2358 25.3171Z"/>';
+
+export interface SLDMetadataNode {
+    id: string;
+    vid: string;
+    nextVId: string;
+    componentType: string;
+    rotationAngle?: number;
+    open: boolean;
+    direction: string;
+    vlabel: boolean;
+    equipmentId: string;
+}
+
+//models just the metadata subelements that are actually used(nodes)
+export interface SLDMetadata {
+    components: unknown[];
+    nodes: SLDMetadataNode[];
+    wires: unknown[];
+    lines: unknown[];
+    arrows: unknown[];
+    layoutParams: unknown;
+}
+
+export type OnNextVoltageCallbackType = (nextVId: string) => void;
+
+export type OnBreakerCallbackType = (
+    breakerId: string,
+    open: boolean,
+    switchElement: EventTarget | null
+) => void;
+
+export type OnFeederCallbackType = (
+    equipmentId: string,
+    equipmentType: string | null,
+    svgId: string,
+    x: number,
+    y: number
+) => void;
+
+export class SingleLineDiagramViewer {
```

Comment:
Part of the constructor, the setters, the getters, init() and draw(), getDimensionsFromSvg() are duplicated from the network-area-diagram. 
Don't we want to try to mutualise them in a parent class ?

Reply:
Discussed with Jon : we'll do it later. Code ok like this

Reply:
Thx, I'll remove the WIP tag, then